bson/src/main/org/bson/codecs/BsonArrayCodec.java

```diff
@@ -21,26 +21,41 @@
 import org.bson.BsonType;
 import org.bson.BsonValue;
 import org.bson.BsonWriter;
+import org.bson.codecs.configuration.CodecProvider;
 import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.configuration.RootCodecRegistry;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 
+import static org.bson.assertions.Assertions.notNull;
+
 /**
  * A codec for BsonArray instances.
  *
  * @since 3.0
  */
 public class BsonArrayCodec implements Codec<BsonArray> {
-    private final CodecRegistry registry;
+    private static final CodecRegistry DEFAULT_REGISTRY = new RootCodecRegistry(Arrays.<CodecProvider>asList(new BsonValueCodecProvider()));
+
+    private final CodecRegistry codecRegistry;
+
+    /**
+     * Creates a new instance with a default {@link org.bson.codecs.configuration.RootCodecRegistry}
+     */
+    public BsonArrayCodec() {
```

Comment:
Follows `BsonDocumentCodec` strictly speaking I'm not sure we need to pass an alternative codecRegistry in.